domain/charm/doc.go

```diff
@@ -0,0 +1,8 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package charm provides the domain types for a charm. The application domain
+// is the primary domain for creation and handling of a charm. The types here
+// are to ensure that the charm is handled correctly and that the charm is
+// correctly represented in the domain.
+package charm
```

Comment:
Would this be better as a sub package under application?
IMO if we are making the call to unify the application domain, leaving artifacts spread across packages when some stuff is moved to application makes things messy.

Reply:
I don't mind it in a subpackage under application, but I want to prefixed with charm, otherwise, the number of types are getting to messy.